packages/cli/src/ui/components/shared/text-buffer.ts

```diff
@@ -33,143 +33,229 @@ function isWordChar(ch: string | undefined): boolean {
   return !/[\s,.;!?]/.test(ch);
 }
 
-// Vim-specific word boundary functions
-export const findNextWordStart = (
-  text: string,
-  currentOffset: number,
-): number => {
-  let i = currentOffset;
+// Helper functions for line-based word navigation
+export const isWordCharStrict = (char: string): boolean => /\w/.test(char);
+
+export const isWhitespace = (char: string): boolean => /\s/.test(char);
+
+// Find next word start within a line, starting from col
+export const findNextWordStartInLine = (
+  line: string,
+  col: number,
+): number | null => {
+  const chars = [...line]; // Handle Unicode properly
```

Comment:
here and elsewhere please use the existing toCodePoints
helper in textUtils. which does the same thing to keep how utf-32 code point handling consistent.